src/Utilities/TypeTraits.hpp

```diff
@@ -1429,3 +1431,29 @@ using function_info = tt_detail::function_info_impl<F>;
 template <typename T, size_t = 0>
 using identity_t = T;
 }  // namespace tt
+
+namespace cpp20 {
+/*!
+ * \ingroup Utilities
+ * Reimplementation of std::swap that is constexpr;
+ * taken from the LLVM source at
+ * https://github.com/llvm-mirror/libcxx/blob/master/include/type_traits
+ */
+template <class T>
+constexpr void swap(T& a, T& b) noexcept {
+  T c(std::move(a));
+  a = std::move(b);
+  b = std::move(c);
+}
+
+/*!
+ * \ingroup Utilities
+ * Reimplementation of std::iter_swap that is constexpr;
+ * taken from the LLVM source at
+ * https://github.com/llvm-mirror/libcxx/blob/master/include/type_traits
+ */
+template <class ForwardIt1, class ForwardIt2>
+constexpr void iter_swap(ForwardIt1 a, ForwardIt2 b) {
```

Comment:
Should be in Algorithm

Reply:
I’ll make these last fixes and rebase on develop when I fix the remaining Travis issues, then I’ll let you know it’s ready for review.